builtin/mainmenu/textures.lua

```diff
@@ -145,6 +145,27 @@ function mm_texture.set_game(identifier,gamedetails)
 	
 	local path = gamedetails.path .. DIR_DELIM .."menu" ..
 									 DIR_DELIM .. identifier .. ".png"
+
+	local count = 1
+	for i = 9, 1, -1 do
+		local check_path = gamedetails.path .. DIR_DELIM .."menu" ..
+					DIR_DELIM .. identifier .. "." .. i .. ".png"
+		local file = io.open(check_path,"rb")
+		if file ~= nil then
+			count = i
+			io.close(file)
+			break
+		end
+	end
+	local new_path = gamedetails.path .. DIR_DELIM .."menu" .. DIR_DELIM ..
+		identifier .. "." .. math.random(1, count) .. ".png"
+	print(new_path)
+	local exists = io.open(new_path,"rb")
+	if exists ~= nil and math.random(1, (count+1)) ~= 1 then
+		io.close(exists)
```

Comment:
Done. (I think the way I did it is the most efficient way possible...)

Also removed the print.